pkg/querier/querier_test.go

```diff
@@ -331,22 +334,22 @@ func TestQuerier(t *testing.T) {
 		Timeout:            1 * time.Minute,
 	}
 	for _, thanosEngine := range []bool{false, true} {
-		var queryEngine v1.QueryEngine
-		if thanosEngine {
-			queryEngine = engine.New(engine.Opts{
-				EngineOpts:        opts,
-				LogicalOptimizers: logicalplan.AllOptimizers,
-			})
-		} else {
-			queryEngine = promql.NewEngine(opts)
-		}
 		for _, query := range queries {
 			for _, encoding := range encodings {
 				for _, streaming := range []bool{false, true} {
 					for _, iterators := range []bool{false, true} {
 						iterators := iterators
 						t.Run(fmt.Sprintf("%s/%s/streaming=%t/iterators=%t", query.query, encoding.name, streaming, iterators), func(t *testing.T) {
 							//parallel testing cause data race
+							var queryEngine v1.QueryEngine
```

Comment:
Why do we need to make this change everywhere? What changed?

Reply:
This change https://github.com/thanos-community/promql-engine/commit/cd3922478f0223daf24deb6aa7669d265ed61045#diff-2e6c4934f63ff9b712c2c346b33036af4724adf70b0801fff9b74f71b37fcd89 caused Go test race detection to complain about calling `engine.New` and `QueryEngine.NewRangeQuery` would cause race condition if running those tests in parallel.